PREGUNTA 1.1:
Examine la base y elabore una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?


In [4]:
install.packages("wooldridge")
library(wooldridge)

set.seed(1)
data(ceosal1)
ceosal1 <- na.omit(ceosal1)
head(ceosal1)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



,salary,pcsalary,sales,roe,pcroe,ros,indus,finance,consprod,utility,lsalary,lsales
,<int>,<int>,<dbl>,<dbl>,<dbl>,<int>,<int>,<int>,<int>,<int>,<dbl>,<dbl>
1,1095,20,27595.0,14.1,106.4,191,1,0,0,0,6.998509,10.225389
2,1001,32,9958.0,10.9,-30.6,13,1,0,0,0,6.908755,9.206132
3,1122,9,6125.9,23.5,-16.3,14,1,0,0,0,7.022868,8.720281
4,578,-9,16246.0,5.9,-25.7,-21,1,0,0,0,6.359574,9.695602
5,1368,7,21783.2,13.8,-3.0,56,1,0,0,0,7.221105,9.988894
6,1145,5,6021.4,20.0,1.0,55,1,0,0,0,7.043160,8.703075


la variables identificadas son:

Variable dependiente:
- salary : salario del CEO

Variables independientes:
- sales: ventas de la empresa
- roe: retornos sobre patrimonio
- ros: retorno sobre ventas

Variables dummies:
- indus
- finance
- consprod
- utility

La base ceosal1 se utiliza para estudiar determinantes del salario de CEOs, especialmente cómo ventas, rentabilidad y tipo de industria afectan el salario ejecutivo.

PREGUNTA 1.2:
Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable.

In [17]:
ceosal1$sales_mayor_promedio <- ifelse(
  ceosal1$sales > mean(ceosal1$sales),
  1,
  0
)

table(ceosal1$sales_mayor_promedio)



  0   1 
155  54 

La variable dummy sales_mayor_promedio toma el valor 1 cuando las ventas de la empresa superan el promedio y 0 en caso contrario.
En la muestra, 54 empresas presentan ventas superiores al promedio, mientras que 155 se encuentran bajo o igual al promedio. Esto permite diferenciar entre empresas grandes y pequeñas para análisis posteriores.

PREGUNTA 1.3:
Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?

In [20]:
ceosal1$sales_mayor_promedio_indus <-
  ceosal1$sales_mayor_promedio * ceosal1$indus

table(ceosal1$sales_mayor_promedio_indus)


  0   1 
188  21 

La interacción entre sales_mayor_promedio e indus permite identificar empresas que pertenecen al sector industrial y que además presentan ventas superiores al promedio. Analiza si las empresas industriales de mayor tamaño presentan un comportamiento diferenciado respecto al resto. toma valor 1 únicamente para empresas del sector industrial con ventas superiores al promedio. En la muestra, 21 empresas cumplen ambas condiciones, mientras que 188 no lo hacen.

PREGUNTA 1.4:
Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?

In [22]:
ceosal1$sales_mayor_promedio_roe <-
  ceosal1$sales_mayor_promedio * ceosal1$roe

head(ceosal1[, c("sales_mayor_promedio", "roe", "sales_mayor_promedio_roe")])

,sales_mayor_promedio,roe,sales_mayor_promedio_roe
,<dbl>,<dbl>,<dbl>
1,1,14.1,14.1
2,1,10.9,10.9
3,0,23.5,0.0
4,1,5.9,5.9
5,1,13.8,13.8
6,0,20.0,0.0


La interacción entre `sales_mayor_promedio` y `roe` indica que el efecto de la rentabilidad sobre el patrimonio (`roe`) solo se considera para aquellas empresas con ventas superiores al promedio.
En los resultados se observa que cuando la dummy toma valor 1, la variable interacción es igual a `roe`, mientras que cuando toma valor 0, el valor es 0. Esto permite analizar si la rentabilidad influye de manera distinta en el salario de los CEOs en empresas grandes en comparación con las de menor tamaño.


PREGUNTA 1.5:
Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.

In [24]:
ceosal1$indus_o_finance <- ifelse(
  ceosal1$indus == 1 | ceosal1$finance == 1,
  1,
  0
)

table(ceosal1$indus_o_finance)


  0   1 
 96 113 

La variable indus_o_finance toma valor 1 cuando la empresa pertenece al sector industrial o financiero, y 0 en caso contrario. En los resultados se observa que 113 empresas pertenecen a alguno de estos dos sectores, mientras que 96 no. Esta variable permite agrupar ambos tipos de empresas para analizar si presentan un comportamiento distinto respecto a las demás industrias.

In [ ]:
2 MODELOS DE REGRESIÓN LINEAL

PREGUNTA 2.1:
La variable indus_o_finance toma valor 1 cuando la empresa pertenece al sector industrial o financiero, y 0 en caso contrario. En los resultados se observa que 113 empresas pertenecen a alguno de estos dos sectores, mientras que 96 no. Esta variable permite agrupar ambos tipos de empresas para analizar si presentan un comportamiento distinto respecto a las demás industrias.

In [25]:
modelo_r <- lm(salary ~ sales + roe + ros, data = ceosal1)
summary(modelo_r)


Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


El modelo analiza cómo el salario del CEO depende de las ventas (sales), la rentabilidad sobre patrimonio (roe) y la rentabilidad sobre ventas (ros). En general, se espera que sales tenga un efecto positivo y significativo, indicando que CEOs de empresas más grandes ganan más. La significancia individual se evalúa con los p-values, mientras que la significancia global se analiza con el estadístico F del modelo.

variable dependiente : salary
variables independientes: sales

PREGUNTA 2.2: Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?

In [26]:
modelo_ur <- lm(
  salary ~ sales + roe + ros +
    sales_mayor_promedio +
    sales_mayor_promedio_roe,
  data = ceosal1
)

summary(modelo_ur)


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)   
(Intercept)              783.298711 258.064395   3.035  0.00272 **
sales                      0.006657   0.011197   0.594  0.55285   
roe                       24.472338  13.088222   1.870  0.06295 . 
ros                       -0.986904   1.457818  -0.677  0.49919   
sales_mayor_promedio     513.083213 510.134319   1.006  0.31572   
sales_mayor_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


La variable sales_mayor_promedio nos dice si pertenecer a empresas con ventas superiores al promedio afecta el salario del CEO, generando un cambio en el nivel base del salario. Por otro lado, la interacción sales_mayor_promedio_roe indica que el efecto de la rentabilidad sobre el patrimonio (roe) no es el mismo para todas las empresas, sino que puede ser distinto en aquellas con ventas altas.

PREGUNTA 2.3: Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%.

In [32]:
F_critico <- qf (0.95 , 2 , 203)
R2_r <- 0.03191
R2_ur <- 0.04059
F_est <- ( R2_ur - R2_r )/(1 - R2_ur )* 203 /2
F_est
F_critico
if ( F_est > F_critico ) {"Se rechaza H0"} else {"No se rechaza H0"}

[1] 0.9182935

[1] 3.040379

[1] "No se rechaza H0"

Como F_est es menor que F_critico, no se rechaza la hipótesis nula. Esto indica que las variables agregadas (sales_mayor_promedio y su interacción con roe) no son conjuntamente significativas al 5%, por lo que no mejoran el modelo de forma relevante.

PREGUNTA 2.4:
Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode.

In [35]:
ceosal1$sales_D <- ceosal1$sales * ceosal1$sales_mayor_promedio
ceosal1$roe_D <- ceosal1$roe * ceosal1$sales_mayor_promedio
ceosal1$ros_D <- ceosal1$ros * ceosal1$sales_mayor_promedio

modelo <- lm(
  salary ~ sales + roe + ros +
    sales_mayor_promedio +
    sales_D + roe_D + ros_D,
  data = ceosal1
)

summary(modelo)

F_critico <- qf(0.95, 4, 201)

R2_ur <- 0.06028
R2_r <- 0.03191

F_est <- ((R2_ur - R2_r) / (1 - R2_ur)) * (201 / 4)

F_est
F_critico
if ( F_est > F_critico ) {"Se rechaza H0"} else {"No se rechaza H0"}


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)  
(Intercept)           331.86179  352.76387   0.941   0.3480  
sales                   0.13362    0.06635   2.014   0.0454 *
roe                    25.63426   13.07068   1.961   0.0512 .
ros                    -0.47570    1.66943  -0.285   0.7760  
sales_mayor_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                -0.13033    0.06731  -1.936   0.0542 .
roe_D                 -17.86857   28.02885  -0.638   0.5245  
ros_D                   1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on 7 and 201

[1] 1.51704

[1] 2.416574

[1] "No se rechaza H0"

Como F_est es menor que F_critico, no se rechaza H0 con un 5% de significancia. Esto indica que no hay evidencia suficiente para afirmar que el comportamiento del modelo es distinto entre empresas con ventas altas y bajas.

PREGUNTA 2.5:
Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales.

In [36]:
ceosal1_indus <- subset(ceosal1, indus == 1)
ceosal1_finance <- subset(ceosal1, finance == 1)
ceosal1_consprod <- subset(ceosal1, consprod == 1)
ceosal1_utility <- subset(ceosal1, utility == 1)

modelo_r <- lm(salary ~ sales + roe + ros, data = ceosal1)

modelo_indus <- lm(salary ~ sales + roe + ros, data = ceosal1_indus)
modelo_finance <- lm(salary ~ sales + roe + ros, data = ceosal1_finance)
modelo_consprod <- lm(salary ~ sales + roe + ros, data = ceosal1_consprod)
modelo_utility <- lm(salary ~ sales + roe + ros, data = ceosal1_utility)

SSR_r <- sum(residuals(modelo_r)^2)

SSR_indus <- sum(residuals(modelo_indus)^2)
SSR_finance <- sum(residuals(modelo_finance)^2)
SSR_consprod <- sum(residuals(modelo_consprod)^2)
SSR_utility <- sum(residuals(modelo_utility)^2)

SSR_ur <- SSR_indus + SSR_finance + SSR_consprod + SSR_utility

G <- 4
k <- 3
n <- nrow(ceosal1)

gl1 <- (G - 1) * (k + 1)
gl2 <- n - G * (k + 1)

F_est <- ((SSR_r - SSR_ur) / gl1) / (SSR_ur / gl2)
F_critico <- qf(0.95, gl1, gl2)

F_est
F_critico
if ( F_est > F_critico ) {"Se rechaza H0"} else {"No se rechaza H0"}


[1] 1.398349

[1] 1.802614

[1] "No se rechaza H0"

Con el test de chow realizado podemos afirmar que no se rechaza H0 con un 5% de significancia, por lo que no hay evidencia suficiente para afirmar que los coeficientes del modelo difieren entre los distintos tipos de empresas. Esto sugiere que la relación entre salario, ventas y rentabilidad es similar entre los sectores analizados.